presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -989,4 +990,51 @@ public void testComplexCast()
                         "SELECT count(DISTINCT a), CAST(max(b) AS VARCHAR) FROM t",
                 "VALUES (1, '0 00:00:01.000')");
     }
+
+    @Test
+    public void testShowSchemasLikeWithEscape()
+    {
+        assertQueryFails("SHOW SCHEMAS IN foo LIKE '%$_%' ESCAPE", "line 1:39: no viable alternative at input '<EOF>'");
+        assertQueryFails("SHOW SCHEMAS LIKE 't$_%' ESCAPE ''", "Escape string must be a single character");
+        assertQueryFails("SHOW SCHEMAS LIKE 't$_%' ESCAPE '$$'", "Escape string must be a single character");
+
+        //make sure that '_' occur in at least one schemaName and not all schemaName contain '_'
+        MaterializedResult allSchemas = computeActual("SHOW SCHEMAS");
+        MaterializedResult subSchemas = computeActual("SELECT * FROM (VALUES 'information_schema','tpch')");
+        assertContains(allSchemas, subSchemas);
+
+        //SHOW SCHEMAS LIKE without escape return all schemaName
+        MaterializedResult notEscape = computeActual("SHOW SCHEMAS LIKE '%_%'");
+        assertEquals(allSchemas, notEscape);
+
+        //SHOW SCHEMAS LIKE with escape only return schemaName contain '_'
+        MaterializedResult withEscape = computeActual("SHOW SCHEMAS LIKE '%$_%' ESCAPE '$'");
+        assertThat(withEscape.getOnlyColumnAsSet()).allMatch(schemaName -> ((String) schemaName).contains("_"));
+    }
+
+    @Test
+    public void testShowTablesLikeWithEscape()
+    {
+        assertQueryFails("SHOW TABLES IN a LIKE '%$_%' ESCAPE", "line 1:36: no viable alternative at input '<EOF>'");
+        assertQueryFails("SHOW TABLES LIKE 't$_%' ESCAPE ''", "Escape string must be a single character");
+        assertQueryFails("SHOW TABLES LIKE 't$_%' ESCAPE '$$'", "Escape string must be a single character");
+
+        String[] testTableName = {"test_escape_1", "test_escape11"};
```

Comment:
`List<String> testTables = ImmutableList.of("test_escape_1", "test_escape11");` would be probably more idiomatic